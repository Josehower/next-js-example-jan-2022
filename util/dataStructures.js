export function getReducedAnimalWithFoods(animalFavoriteFoods) {
  const animalWithFoods = {
    id: animalFavoriteFoods[0].animalId,
    firstName: animalFavoriteFoods[0].animalFirstName,
    age: animalFavoriteFoods[0].animalAge,
    type: animalFavoriteFoods[0].animalType,
    accessory: animalFavoriteFoods[0].animalAccessory,
    foods: animalFavoriteFoods.map((animalFavoriteFood) => {
      return {
        id: animalFavoriteFood.foodId,
        name: animalFavoriteFood.foodName,
        type: animalFavoriteFood.foodType,
      };
    }),
  };

  return animalWithFoods;
}
