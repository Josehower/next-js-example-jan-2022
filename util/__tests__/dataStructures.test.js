import { getReducedAnimalWithFoods } from '../dataStructures';

test('reduces animal favorite foods', () => {
  const animalFavoriteFoods = [
    {
      animalId: 2,
      animalFirstName: 'Pete',
      animalAge: 4,
      animalType: 'Iguana',
      animalAccessory: 'Top Hat',
      foodId: 3,
      foodName: 'Cupcakes',
      foodType: 'Cake',
    },
    {
      animalId: 2,
      animalFirstName: 'Pete',
      animalAge: 4,
      animalType: 'Iguana',
      animalAccessory: 'Top Hat',
      foodId: 4,
      foodName: 'Burritos',
      foodType: 'Mexican',
    },
  ];

  expect(getReducedAnimalWithFoods(animalFavoriteFoods)).toStrictEqual({
    id: 2,
    firstName: 'Pete',
    age: 4,
    type: 'Iguana',
    accessory: 'Top Hat',
    foods: [
      { id: 3, name: 'Cupcakes', type: 'Cake' },
      { id: 4, name: 'Burritos', type: 'Mexican' },
    ],
  });

  // Jest snapshots may be helpful if you are
  // writing a lot of data structures by hand
  expect(getReducedAnimalWithFoods(animalFavoriteFoods)).toMatchSnapshot();
});
