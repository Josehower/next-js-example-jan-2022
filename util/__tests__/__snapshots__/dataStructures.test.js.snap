// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reduces animal favorite foods 1`] = `
Object {
  "accessory": "Top Hat",
  "age": 4,
  "firstName": "Pete",
  "foods": Array [
    Object {
      "id": 3,
      "name": "Cupcakes",
      "type": "Cake",
    },
    Object {
      "id": 4,
      "name": "Burritos",
      "type": "Mexican",
    },
  ],
  "id": 2,
  "type": "Iguana",
}
`;
